### PR TITLE
Fix `rpc` call from name based `Callable`s

### LINKED ISSUE #87841: Callable.rpc() on RPC function throws an error "Invalid call. Nonexistent function 'rpc' in base 'Callable'.
### Tested versions

- Reproducible in: Godot v4.2.1 stable win64

### System information

Windows 10 - Godot v4.2.1.stable

### Issue description

Creating a `Callable` and calling the `rpc` function throws an error stating that the call is invalid, but calling the `rpc` function on the function and not the `Callable` works.

### Steps to reproduce

Create a new GDScript in a blank project, attach it to a node, add the following to the script:
```gdscript
@rpc
func hello():
	print("RPC fn has been called!")

func _ready():
	hello.rpc() # OK
	var fn = Callable(self, "hello")
	fn.rpc() # ERR
```

### Minimal reproduction project (MRP)

[bugreport1.zip](https://github.com/godotengine/godot/files/14131316/bugreport1.zip)

### PR DESCRIPTION
**Pull request by AThousandShips**
Makes `rpc` calls work for the `Callable(obj, method)` form `Callable`s, works like the `GDScriptRPCCallable` call (which also does no special treatment for the error return, I'll leave it as such and leave any improvements or changes to that side to future improvements to both)

* Fixes: https://github.com/godotengine/godot/issues/87841

### CODE REVIEW

File: core/variant/callable.cpp [hunk 1]
```diff
@@ -91,10 +92,17 @@ Error Callable::rpcp(int p_id, const Variant **p_arguments, int p_argcount, Call
 		r_call_error.expected = 0;
 		return ERR_UNCONFIGURED;
 	} else if (!is_custom()) {
-		r_call_error.error = CallError::CALL_ERROR_INVALID_METHOD;
-		r_call_error.argument = 0;
-		r_call_error.expected = 0;
-		return ERR_UNCONFIGURED;
+		Node *node = Object::cast_to<Node>(ObjectDB::get_instance(ObjectID(object)));
+#ifdef DEBUG_ENABLED
```
Comment (AThousandShips) [inline core/variant/callable.cpp:96]: Using the same debug only check here

File: core/variant/callable.cpp [hunk 2]
```diff
@@ -35,6 +35,7 @@
 #include "core/object/script_language.h"
 #include "core/variant/callable_bind.h"
 #include "core/variant/variant_callable.h"
+#include "scene/main/node.h"
```
Comment (akien-mga) [inline core/variant/callable.cpp:38]: This is problematic, `core` shouldn't depend on `scene`.
There are a couple exceptions currently that we need to fix, but in theory `core` shouldn't even know about Node.

Comment (AThousandShips) [reply core/variant/callable.cpp:38]: That's true, but RPC calls simply require `Node`, so unsure how to solve otherwise

Comment (akien-mga) [reply core/variant/callable.cpp:38]: See #53295 for context.

Comment (akien-mga) [reply core/variant/callable.cpp:38]: You can likely work it around with strings:
```cpp
Object *obj = ObjectDB::get_instance(ObjectID(object));
#ifdef DEBUG_ENABLED
if (!obj || !obj->is_class("Node")) {
	r_call_error.error = CallError::CALL_ERROR_INSTANCE_IS_NULL;
	r_call_error.argument = 0;
	r_call_error.expected = 0;
	return ERR_UNCONFIGURED;
}
#endif
r_call_error.error = CallError::CALL_OK;
return obj->call("rpcp", p_id, method, p_arguments, p_argcount);
```

Comment (AThousandShips) [reply core/variant/callable.cpp:38]: Sounds good, will look at that!

File: core/variant/callable.cpp [hunk 3]
```diff
@@ -92,10 +92,24 @@ Error Callable::rpcp(int p_id, const Variant **p_arguments, int p_argcount, Call
...
+#endif
+		LocalVector<const Variant *> argptrs;
+		argptrs.resize(p_argcount + 2);
+		Variant args[2] = { p_id, method };
+		argptrs[0] = &args[0];
+		argptrs[1] = &args[1];
+		for (int i = 0; i < p_argcount; ++i) {
+			argptrs[i + 2] = p_arguments[i];
+		}
+		return (Error)obj->callp("rpc_id", argptrs.ptr(), p_argcount + 2, r_call_error).operator int64_t();
```
Comment (AThousandShips) [inline core/variant/callable.cpp:112]:
```suggestion
		int argcount = p_argcount + 2;
		const Variant **argptrs = (const Variant **)alloca(sizeof(Variant *) * argcount);
		Variant args[2] = { p_id, method };
		argptrs[0] = &args[0];
		argptrs[1] = &args[1];
		for (int i = 0; i < p_argcount; ++i) {
			argptrs[i + 2] = p_arguments[i];
		}
		return (Error)obj->callp("rpc_id", argptrs, argcount, r_call_error).operator int64_t();
```
Alternatively

File: core/variant/callable.cpp [hunk 4]
```diff
@@ -92,10 +92,31 @@ Error Callable::rpcp(int p_id, const Variant **p_arguments, int p_argcount, Call
...
+		const Variant args[2] = { p_id, method };
+
+		argptrs[0] = &args[0];
+		argptrs[1] = &args[1];
+		for (int i = 0; i < p_argcount; ++i) {
+			argptrs[i + 2] = p_arguments[i];
+		}
+
+		CallError tmp;
+		Error err = (Error)obj->callp("rpc_id", argptrs, argcount, tmp).operator int64_t();
```
Comment (KoBeWi) [inline core/variant/callable.cpp:116]:
```suggestion
		Error err = (Error)obj->callp(SNAME("rpc_id"), argptrs, argcount, tmp).operator int64_t();
```

**Review by KoBeWi — APPROVED** [review]
This fixes the error, but not sure how to confirm whether the method is actually called.
The code looks fine though.

**Review by Faless — APPROVED** [review]
This looks great to me!

I remember @reduz wanted this to be implemented at the language level using custom callables, but I think it make sense to also have the option of using `Callable(obj, method)` as an alternative if the language doesn't implement it.

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank *you*!

**Pull request merged**